R/createRecodeDB.R

```diff
@@ -1,38 +1,77 @@
+#' @param directory Path of the directory where the data base will be saved.
+#' @param DBname Name of the database (will be used as filename).
+#' @param fileType "csv2" (default), "csv", "xlsx"
 #'
 #' @return NULL
 #'
-#'
 #' @examples
 #' # create a named list of data frames
-#' recodeListList <- list( Europe = data.frame(
+#' Countries <- list( Europe = data.frame(
 #'                                   id = 1:4,
 #'                                   oldValues = c("Berlin", "Copenhagen", "Rome", "Madrid"),
 #'                                   newValues = c("Germany", "Denmark", "Italy" , "Spain")),
 #'                         Asia = data.frame(
 #'                                   id = 1:4,
 #'                                   oldValues = c("Baku", "Tokyo", "Kathmandu", "Singapore"),
 #'                                   newValues = c("Azerbaijan", "Japan", "Nepal" , "Singapore")))
-#' recodeListList
-#' filePath_temp <- tempfile(fileext = ".xlsx")
-#' # create the Excel-file
-#' createRecodeDB(recodeListList = recodeListList, filePath = filePath_temp)
+#' Countries
+#' directory <- tempdir()
+#' # create .csv file with comma as decimal point (default)
+#' createRecodeDB(recodeListList = Countries,
+#'                directory = directory,
+#'                DBname = "Countries",
+#'                fileType = "csv2")
+#' # create .csv file
+#' createRecodeDB(recodeListList = Countries,
+#'                directory = directory,
+#'                DBname = "Countries",
+#'                fileType = "csv")
+#' # create .xlsx file
+#' createRecodeDB(recodeListList = Countries,
+#'                directory = directory,
+#'                DBname = "Countries",
+#'                fileType = "xlsx")
 #'
 #' @export
-#'
-createRecodeDB <- function(recodeListList, filePath) {
+createRecodeDB <- function(recodeListList, directory = getwd(), DBname, fileType = "csv2") {
+
+  # checks ---------------------------------------------------------------------
   if (!is.list(recodeListList)) stop("'recodeListList' must be a named list of data.frames.")
   if (!is.data.frame(recodeListList[[1]])) stop("'recodeListList' must be named a list of data.frames.")
   if (is.null(names(recodeListList)) || any(is.na(names(recodeListList)))) stop("'recodeListList' must be named a list of data.frames.")
   lapply(recodeListList, checkRecodeList)
 
-  writexl::write_xlsx(recodeListList, path = filePath, col_names = TRUE)
+  # xlsx files -----------------------------------------------------------------
+  if(fileType == "xlsx") {
+    writexl::write_xlsx(recodeListList,
+                        path = paste0(directory,"\\",DBname,".xlsx"),
+                        col_names = TRUE)
+    return(NULL) }
+
+  # csv files ------------------------------------------------------------------
+  dirpath <- file.path(directory,
+                       DBname,
+                       fsep = "\\")
+  unlink(dirpath, recursive = TRUE) # delete existing file ("overwrite")
```

Comment:
@apelalina So I've given it some thought, and I think the harm that could be done when overwriting by default can be quite severe: Suppose I have a directory of databases and want to create a new one, but don't check if the name already exists. If I create a new database without a warning by the function, I would run into the problem that my old, already filled database just gets overwritten without me not even noticing. Do you think that could be a valid problem?
Therefore I would prefer the default for `overwrite = FALSE`.

Reply:
@nickhaf I agree. To avoid confusion with the parameter `override` in `updateRecodeDB` (Should existing recode pairs be overwritten when conflicting newer recode pairs?), I suggest renaming `override` in `updateRecodeDB` to `replace`.